common/src/main/java/com/revenuecat/purchases/common/caching/DateExtensions.kt

```diff
@@ -0,0 +1,33 @@
+package com.revenuecat.purchases.common.caching
+
+import com.revenuecat.purchases.common.DateProvider
+import com.revenuecat.purchases.common.DefaultDateProvider
+import com.revenuecat.purchases.common.LogIntent
+import com.revenuecat.purchases.common.log
+import com.revenuecat.purchases.strings.ReceiptStrings
+import java.util.Date
+import kotlin.time.Duration
+import kotlin.time.Duration.Companion.hours
+import kotlin.time.Duration.Companion.milliseconds
+import kotlin.time.Duration.Companion.minutes
+
+private val CACHE_REFRESH_PERIOD_IN_FOREGROUND = 5.minutes
+private val CACHE_REFRESH_PERIOD_IN_BACKGROUND = 25.hours
+
+internal fun Date?.isStale(appInBackground: Boolean, dateProvider: DateProvider = DefaultDateProvider()): Boolean {
```

Comment:
This name is a bit weird since it's in `Date`, so it has no indication that it's related to caches. Now that it's exposed as `internal` I'd make it a bit more clear, maybe:
```suggestion
internal fun Date?.isCacheStale(appInBackground: Boolean, dateProvider: DateProvider = DefaultDateProvider()): Boolean {
```